### PR TITLE
CPM-1005 - CPM-1006: Add Repository for Simple Select nomenclature

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [X] Tests

### CODE REVIEW

File: components/identifier-generator/back/tests/Acceptance/Context/SimpleSelectNomenclatureContext.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+class SimpleSelectNomenclatureContext implements Context
+{
+    private ?ViolationsException $violations = null;
+    private const DEFAULT_OPERATOR = '<=';
+    private const DEFAULT_VALUE = 3;
+    private const DEFAULT_GENERATE_IF_EMPTY = false;
+
+    public function __construct(
+        private readonly SimpleSelectNomenclatureRepository $nomenclatureRepository,
+        private readonly UpdateNomenclatureHandler    $updateNomenclatureValuesHandler,
```
Comment (antoinearrive) [inline components/identifier-generator/back/tests/Acceptance/Context/SimpleSelectNomenclatureContext.php:24]:
```suggestion
        private readonly UpdateNomenclatureHandler $updateNomenclatureValuesHandler,
```

File: components/identifier-generator/back/tests/EndToEnd/Infrastructure/Controller/UpdateSimpleSelectNomenclatureControllerEndToEnd.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        Assert::assertSame(Response::HTTP_BAD_REQUEST, $response->getStatusCode());
+    }
+
+    protected function getConfiguration(): Configuration
+    {
+        return $this->catalog->useTechnicalCatalog(['identifier_generator']);
+    }
+
+    private function getNomenclatureRepository(): SimpleSelectNomenclatureRepository
+    {
```
Comment (antoinearrive) [inline components/identifier-generator/back/tests/EndToEnd/Infrastructure/Controller/UpdateSimpleSelectNomenclatureControllerEndToEnd.php:122]:
```suggestion
    {SimpleSelectNomenclatureRepository::class ?
```

**Review by antoinearrive — APPROVED** [review]

File: components/identifier-generator/back/src/Application/Get/GetNomenclatureHandler.php [hunk 3]
```diff
@@ -29,7 +31,13 @@ public function __construct(
      */
     public function __invoke(GetNomenclatureCommand $command): array
     {
-        $nomenclature = $this->nomenclatureRepository->get($command->propertyCode()) ?? new NomenclatureDefinition();
+        if ($command->propertyCode() === 'family') {
```
[the remark this answers is not in the document]
Comment (Balamung) [reply components/identifier-generator/back/src/Application/Get/GetNomenclatureHandler.php:34]: Yep ! Used the constant at the end of the PR and didn't go back and update everywhere :D

**Review by mmetayer — APPROVED** [review]
GTM after wrapping SQL transactions

**Pull request merged**